modules/envoy-gateway/create-synapse-oauth-client.py

```diff
@@ -0,0 +1,26 @@
+import synapseclient
+import json
+syn = synapseclient.login()
```

Comment:
We might be able to create a client for each service that communicates with the cluster. Thus offloading the responsibility to get an auth token on the caller. Then - in the envoy gateway, we have a JWT auth provider with 1 or more providers to verify the signature.

Reply:
Oh interesting - for every service, we could auto-create an oauth client? Not sure if we've tested this out explicitly with envoy gateway, or if thats another ticket.

Reply:
For our use-case with machine-to-machine authentication or the `client credentials grant type from OAuth` we cannot use Synapse as it is not supported.

What we can use is Auth0 for this purpose. I did a small PoC that I could generate and auth an Auth0 token, but I hadn't set up the full round-trip for it, I also want to test out the extension part of the otel exporter as this is what Schematic would need to do on their side: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/oauth2clientauthextension